[TECH] Supprimer ember-api-actions de Pix Orga (PIX-20768)

Libouk opened the pull request:
## ❄️ Problème
L’utilisation de Ember Actions n’est pas nécessaire et pas justifiée pour faire des requêtes non-liées à des ressources.

## 🛷 Proposition

1. Corriger la route `/oidc/user/check-reconciliation` dans Mirage JS qui renvoie un `201` alors qu’elle renvoie un `200` dans Pix API (ce qui fausse la compréhension quand on écrit des tests)
2. Utiliser un Fetch plutôt que les Member Actions. =

## 🧑‍🎄 Pour tester

### Test de la confirmation d’association OIDC (`@onSubmit={{@controller.redirectToAssociationConfirmation}}`)

1. Lancer Pix Orga
2. Créer une invitation à rejoindre une organisation sur Pix Orga
3. Se connecter via SSO
4. Sur la page `pour le (votre compte Pix) lier à votre nouvelle méthode de connexion`, entrer une adresse e-mail ou un mot de passe ne correspondant pas à un compte existant
5. Constater que s’affiche le message `L'adresse e-mail ou l'identifiant et/ou le mot de passe saisis sont incorrects.`
6. Sur la page `pour le (votre compte Pix) lier à votre nouvelle méthode de connexion`, cette fois entrer une adresse e-mail et un mot de passe correspondant pas à un compte existant
7. Constater que la page de confirmation d'association s'affiche avec le nom complet, nom et prénom et adresse email, vérifier dans le session storage la présence des informations sur l'utilisateur SSO

### Test de la connexion par email et mot de passe (`@onSubmit={{@controller.authenticate}}`)

1. Aller sur la page de login par _« Adresse e-mail »_  et _« Mot de passe »_ (https://orga.dev.pix.fr/connexion) et spécifier l’adresse email d’un utilisateur existant ayant accès à une organisation (par exemple `james-paledroits@example.net`) mais spécifier un mauvais mot de passe et effectuer des tentatives de connexion avec ce mauvais mot de passe
2. Au bout d’un moment le message suivant doit s’afficher _« Le message qui s'affiche : Vous avez effectué trop de tentatives de connexion, votre compte Pix est bloqué temporairement pendant 2 minutes. Réessayez plus tard ou cliquez sur [mot de passe oublié](https://orga.dev.pix.fr/mot-de-passe-oublie) pour le réinitialiser. »_

Libouk committed:
refactor(orga): remove useless ember data model

Libouk committed:
refactor(orga): remove now useless ember actions package

pix-bot-github commented:
Choisir les applications à déployer :

- [x] [API](https://api-pr14398.review.pix.fr/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr14398/environment) <!-- pix-api-review -->
- [x] [App (.fr)](https://app-pr14398.review.pix.fr) – [App (.org)](https://app-pr14398.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-app-review-pr14398/environment) <!-- pix-app-review -->
- [x] [Orga (.fr)](https://orga-pr14398.review.pix.fr) – [Orga (.org)](https://orga-pr14398.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-orga-review-pr14398/environment) <!-- pix-orga-review -->
- [ ] [Certif (.fr)](https://certif-pr14398.review.pix.fr) – [Certif (.org)](https://certif-pr14398.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-certif-review-pr14398/environment) <!-- pix-certif-review -->
- [ ] [Junior](https://junior-pr14398.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-junior-review-pr14398/environment) <!-- pix-junior-review -->
- [ ] [Admin](https://admin-pr14398.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-admin-review-pr14398/environment) <!-- pix-admin-review -->
- [ ] [API MaDDo](https://pix-api-maddo-review-pr14398.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr14398/environment) <!-- pix-api-maddo-review -->
- [ ] [Audit Logger](https://pix-audit-logger-review-pr14398.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr14398/environment) <!-- pix-audit-logger-review -->

> [!IMPORTANT]
> N'oubliez pas de déployer l'API pour pouvoir accéder aux fronts et/ou à l’API MaDDo.

Libouk commented:
Func OK
<img width="1597" height="922" alt="image" src="https://github.com/user-attachments/assets/8d731610-b365-41a7-9357-98e07ab87bf6" />

Libouk commented on orga/app/controllers/authentication/oidc/login.js:
```diff
@@ -26,15 +27,35 @@ export default class OidcLoginController extends Controller {
 
-    const oidcAssociationConfirmationData = await authenticationRequest.login();
-    oidcAssociationConfirmationStorage.set({ ...oidcAssociationConfirmationData, email });
+    const json = await response.json();
```
```suggestion: 
const reconciliationJsonData = await response.json();
```
Pour avoir un nom plus parlant quant au contenu de ce json

EmmanuelleBonnemay replied:
ok!

lego-technix commented on orga/app/controllers/authentication/oidc/login.js:
```diff
@@ -26,15 +27,35 @@ export default class OidcLoginController extends Controller {
-    oidcAssociationConfirmationStorage.set({ ...oidcAssociationConfirmationData, email });
+    const userOidcReconciliationJson = await response.json();
+
+    if (!response.ok) throw { isAdapterError: true, ...userOidcReconciliationJson };
```
La gestion du cas d’une réponse en erreur n’est pas correcte. Il faudrait corriger comme suit, car il faut gérer le cas d’erreur avant de chercher à consommer la payload de la response avec `response.json()`.

On se rend ainsi compte qu’il n’y a pas de test, ou un test testant mal, le cas d’une réponse en erreur.

lego-technix replied:
✅ FAIT Gestion d’erreur retravaillée.

lego-technix committed:
test(orga): fix mirage route /oidc/user/check-reconciliation

to behave like the API, by having a status 200 instead of 201 (mirage default
status for POST routes).

lego-technix committed:
test(orga): use translation key instead of translated strings

as stated by Pix coding standards.

lego-technix committed:
test(mon-pix): fix mirage route /oidc/user/check-reconciliation

to behave like the API, by having a status 200 instead of 201 (mirage default
status for POST routes).

lego-technix committed:
refactor(orga): fix and simplify error handling

by considering the 3 possible cases in this context:
* EmberSimpleAuth
* JSON:API response
* non-JSON:API response

lego-technix committed:
refactor(orga): use fetch instead of member actions in association check

lego-technix committed:
test(orga): improve test context and title of OIDC login test

lego-technix committed:
test(orga): add tests for errors in OIDC authentication flow

lego-technix committed:
test(orga): improve test context of login test

EmmanuelleBonnemay commented:
Le message qui s'affiche : Vous avez effectué trop de tentatives de connexion, votre compte Pix est bloqué temporairement pendant 2 minutes. Réessayez plus tard ou cliquez sur [mot de passe oublié](https://orga.dev.pix.fr/mot-de-passe-oublie) pour le réinitialiser.

lego-technix commented:
> Le message qui s'affiche : Vous avez effectué trop de tentatives de connexion, votre compte Pix est bloqué temporairement pendant 2 minutes. Réessayez plus tard ou cliquez sur [mot de passe oublié](https://orga.dev.pix.fr/mot-de-passe-oublie) pour le réinitialiser.

Effectivement c’est bien le message qui s’affiche car il y a déjà un blocage temporaire avant un blocage permanent. Je vais mettre la description de la PR à jour. De mon côté j’avais testé jusqu’au blocage définitif.

lego-technix commented:
> > Le message qui s'affiche : Vous avez effectué trop de tentatives de connexion, votre compte Pix est bloqué temporairement pendant 2 minutes. Réessayez plus tard ou cliquez sur [mot de passe oublié](https://orga.dev.pix.fr/mot-de-passe-oublie) pour le réinitialiser.
> 
> Effectivement c’est bien le message qui s’affiche car il y a déjà un blocage temporaire avant un blocage permanent. Je vais mettre la description de la PR à jour. De mon côté j’avais testé jusqu’au blocage définitif.

✅ Mise à jour de la description de la PR FAITE.

EmmanuelleBonnemay reviewed (approved):
Lu et testé avec succès

pix-service commented:
:tada: This PR is included in version 5.282.0 :tada:

The release is available on [GitHub release](https://github.com/1024pix/pix/releases/tag/v5.282.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
